MobileCenter/MobileCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -185,10 +202,26 @@ - (void)deleteLogsFromDBWithColumnValues:(NSArray<NSString *> *)columnValues col
 
   // Execute.
   if ([self.connection executeQuery:deleteLogsQuery]) {
-    MSLogVerbose([MSMobileCenter logTag], @"%@ %@", deletionTrace, @"succeded");
+    MSLogVerbose([MSMobileCenter logTag], @"%@ %@", deletionTrace, @"succeeded");
   } else {
     MSLogError([MSMobileCenter logTag], @"%@ %@", deletionTrace, @"failed");
   }
 }
 
+- (void)deleteOldestLogWithGroupId:(NSString *)groupId {
+  NSString *deleteLogQuery =
+      [NSString stringWithFormat:@"DELETE FROM %@ WHERE %@ = '%@' ORDER BY %@ ASC LIMIT 1", kMSLogTableName,
```

Comment:
Currently it will recover from a failed deletion by redeleting one by one on each other logs but yes I agree we can directly remove all logs exceeding the limit, that's much quicker, good idea.